### DISCUSSION TITLE
An inline search-tool for different type of blocks (similar to notion)

### DISCUSSION THREAD

**Discussion by SargeKhan** [codex-team/editor.js#1922] [Q&A]
### The question:
Looking at the API docs, the tools API only allows opening/closing the toolbar and the inline tools can only work when text is selected. I am curious to know if we can have search based in line tool for creating various different kinds of blocks. 
### Why and how the question has come up:
Notion's commands tool is quite useful.

**Comment by tvvignesh** [comment]
@gohabereg @neSpecc Very much interested in this since it will make it a lot usable for users rather than having to click the plus icon and look for the tool every time. May I know if we can somehow extend editor.js and implement this similar to notion's command search popup? 

I have thought of a way of doing it but does not seem clean.

1) I can listen for the keypress using https://editorjs.io/listeners
2) On the keypress (`/` in this case), I can open up a modal/dropdown from JS for listing all blocks
3) Since I don't find an API for searching or listing blocks, I would have to implement that myself within the modal
4) Once they select something, I add to the canvas using https://editorjs.io/blocks#insert

Would this work? Thanks.
